sdk/eventhubs/azure-messaging-eventhubs/test/ut/test_checkpoint_store.hpp

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include <azure/core/context.hpp>
+#include <azure/core/internal/environment.hpp>
+#include <azure/identity.hpp>
+#include <azure/messaging/eventhubs.hpp>
+
+#include <gtest/gtest.h>
+
+namespace Azure { namespace Messaging { namespace EventHubs { namespace Test {
+  class TestCheckpointStore : public Azure::Messaging::EventHubs::CheckpointStore {
+    std::map<std::string, Azure::Messaging::EventHubs::Models::Checkpoint> m_checkpoints;
+    std::map<std::string, Azure::Messaging::EventHubs::Models::Ownership> m_ownerships;
+
+  public:
+    TestCheckpointStore()
+    {
+      m_checkpoints = std::map<std::string, Azure::Messaging::EventHubs::Models::Checkpoint>();
+      m_ownerships = std::map<std::string, Azure::Messaging::EventHubs::Models::Ownership>();
+    }
+
+    std::vector<Azure::Messaging::EventHubs::Models::Checkpoint> ListCheckpoints(
+        std::string const& fullyQualifiedNamespace,
+        std::string const& eventHubName,
+        std::string const& consumerGroup,
+        ListCheckpointsOptions options = ListCheckpointsOptions(),
+        Azure::Core::Context ctx = Azure::Core::Context()) override
+    {
+      (void)fullyQualifiedNamespace;
+      (void)eventHubName;
+      (void)consumerGroup;
+      (void)ctx;
+      (void)options;
+      std::vector<Azure::Messaging::EventHubs::Models::Checkpoint> checkpoints;
+      for (auto const& checkpoint : m_checkpoints)
+      {
+        checkpoints.push_back(checkpoint.second);
+      }
+      return checkpoints;
+    }
+
+    std::vector<Azure::Messaging::EventHubs::Models::Ownership> ListOwnership(
+        std::string const& fullyQualifiedNamespace,
+        std::string const& eventHubName,
+        std::string const& consumerGroup,
+        ListOwnershipOptions options = ListOwnershipOptions(),
+        Azure::Core::Context ctx = Azure::Core::Context()) override
+    {
+      (void)fullyQualifiedNamespace;
+      (void)eventHubName;
+      (void)consumerGroup;
+      (void)ctx;
+      (void)options;
+      std::vector<Azure::Messaging::EventHubs::Models::Ownership> ownerships;
+      for (auto const& ownership : m_ownerships)
+      {
+        ownerships.push_back(ownership.second);
+      }
+      return ownerships;
+    }
+
+    std::vector<Azure::Messaging::EventHubs::Models::Ownership> ClaimOwnership(
+        std::vector<Models::Ownership> partitionOwnership,
+        ClaimOwnershipOptions const& options = ClaimOwnershipOptions(),
+        Azure::Core::Context ctx = Azure::Core::Context()) override
```

Comment:
There is a ton of places in this PR where
1. Context is passed by value
2. Has fully qualified namespace
3. The name is `ctx` and not `context`
4. Default value is written unnecessarily verbosely (`Azure::Core::Context()` instead of `{}`).

I am only leaving a single comment, but please, correct them all. Note - there are places where context may be passed by ref, but still has the fully qualified namespace and is called `ctx`.

```suggestion
        Context const& context = {}) override
```